server/server.go

```diff
@@ -786,7 +790,7 @@ func (s *Server) createClient(conn net.Conn) *client {
 		if err := conn.Handshake(); err != nil {
 			c.Errorf("TLS handshake error: %v", err)
 			c.sendErr("Secure Connection - TLS Required")
-			c.closeConnection()
+			c.closeConnection(TLSHandshakeError)
```

Comment:
Humm... I tried to connect a non TLS client to a TLS server and I get Write Error, not TLS Handshake Error. I believe that since we call sendErr(), it is possible that flushOutbound() gets an error because the client will also close the connection (after receiving the error), so the actual close of the connection occurs there, with WriteError. That may then not prove that useful.

Reply:
Non TLS client won't try to handshake will it? I will look into it though, agree we want the correct reason.

Reply:
No they won't.

Actually, what happened is that I was connecting without TLS, so the client when getting the INFO fails the connect and says that secure connection is required. The server when getting into the Handshake will fail and send the error, but since the connection is closed it will get a write error, so that kind of makes sense.

But, I just tried with a client connection with TLS but without the CA root cert so it won't be able to verify the server certificate, and I still get the Write Error as a reason.

Reply:
ok I will fix and add a test, thanks.